pubtools/_quay/operator_pusher.py

```diff
@@ -215,46 +249,32 @@ def iib_add_bundles(self, bundles, archs, ocp_version, deprecation_list=None):
                 External URLs to bundle images to be added to the index image.
             archs ([str]):
                 Architectures to build for.
-            ocp_version (str):
-                OCP version to add the bundles to. It acts as a tag of the index image.
+            index_image (str):
+                Index image to add the bundles to.
             deprecation_list ([str]):
                 List of bundles to be deprecated.
+            target_settings (dict):
+                Settings used for setting the value of pubtools-iib parameters.
 
         Returns (dict):
             Build details provided by IIB.
         """
         LOG.info(
-            "Requesting IIB to add bundles '{0}' to index image version '{1}'".format(
-                bundles, ocp_version
-            )
-        )
-        args = ["--skip-pulp"]
-
-        args += ["--iib-server", self.target_settings["iib_server"]]
-        args += ["--iib-krb-principal", self.target_settings["iib_krb_principal"]]
-
-        if "iib_overwrite_from_index" in self.target_settings:
-            args += ["--overwrite-from-index"]
-        if "iib_krb_ktfile" in self.target_settings:
-            args += ["--iib-krb-ktfile", self.target_settings["iib_krb_ktfile"]]
-
-        index_image = "{image_repo}:{tag}".format(
-            image_repo=self.target_settings["iib_index_image"], tag=ocp_version
+            "Requesting IIB to add bundles '{0}' to index image '{1}'".format(bundles, index_image)
         )
-        args += ["--index-image", index_image]
-        for bundle in bundles:
-            args += ["--bundle", bundle]
-        for arch in archs:
-            args += ["--arch", arch]
+        args, env_vars = cls.pubtools_iib_get_common_args(target_settings)
+
+        if index_image:
+            args += ["--index-image", index_image]
+        if bundles:
+            for bundle in bundles:
+                args += ["--bundle", bundle]
+        if archs:
+            for arch in archs:
+                args += ["--arch", arch]
         # inconsistent way of presenting multiple arguments...
         if deprecation_list:
-            args += ["--deprecation-list", ",".join(deprecation_list)]
-
-        env_vars = {}
-        if "iib_overwrite_from_index_token" in self.target_settings:
-            env_vars["OVERWRITE_FROM_INDEX_TOKEN"] = self.target_settings[
-                "iib_overwrite_from_index_token"
-            ]
+            args += ["--deprecation-list", deprecation_list]
```

Comment:
This argument expects a string.

Reply:
The value is taken from pub. Pub doesn't parse it so it's only a comma separated string.

Reply:
It's taken from pub for iib-add-bundles entrypoint. But for push_docker, it's from get_deprecation_list which returns a list.
```
            deprecation_list ([str]):
                List of bundles to be deprecated.
```
This is also incorrect if it's expected to be a string.

Reply:
You're right, fixed.